repos/spack_repo/builtin/packages/nlohmann_json/package.py

```diff
@@ -55,6 +55,25 @@ class NlohmannJson(CMakePackage):
     conflicts("%gcc@:4.8", when="@:3.2.9")
     conflicts("%intel@:16")
 
+    def patch(self):
+        if self.spec.satisfies("@3.11.0:3.12.0"):
```

Comment:
Maybe better to define the patch routine only when it applies:
```suggestion
    @when("@3.11.0:3.12.0")
    def patch(self):
```
or provide an actual patch file for the diff, or pull in the PR you mention?

Is the motivation a wide range of compatibility with nljson versions?

Reply:
> Maybe better to define the patch routine only when it applies

I have never really looked into how when decorators apply when they are used with overlapping versions ranges but retain the same function name... Doesn't apply here, but that's why I've been placing the satisfies inside the patch methods.

Reply:
> Is the motivation a wide range of compatibility with nljson versions?

Motivation is only realistically allowing modules with the last version. But this also applies cleanly to older versions.

Reply:
> or pull in the PR you mention?

Too much crud included.